code/renderers/vue3/src/render.ts

```diff
@@ -50,15 +52,38 @@ export function renderToCanvas(
       Use "() => ({ template: '<my-comp></my-comp>' })" or "() => ({ components: MyComp, template: '<my-comp></my-comp>' })" when defining the story.
       `,
     });
-    return;
+    return () => {};
   }
 
-  showMain();
+  const storyArgs = element.props || (element as any).render?.().props || storyContext.args || {};
```

Comment:
Is it safe to call `render()` here? Or can it have side-effects, that would be triggered multiple times by this call? I'm no Vue expert

Reply:
well anyway in Vue 3 render function arguments consistent in stateful and functional components, using h().
so if i want to render any component  i will have to 
  ```
  const FunctionalComp = props => h('button', `I'm functional ${props.label}`)
  const render = () =>  h(FunctionalComp, { label: 'Hello Story' }, [])
  const { type , props, children, key } = render()
  // pure  
  const pureRenderFn = props => h(MyButton,props)
  const { type : t , props : p, children : c , key:k } = pureRenderFn({label:'Im pure '})
```
i wrote this code [LINK](https://sfc.vuejs.org/#eNqlVF1r2zAU/SsXM1ACjv2yJ+MUtkFZB/tgfawLdRw5dmtLQrrOFoz/+64kK3Fa2EsxxpZ0dO45V/dqjD4plRwHHmVRIXJT6VYhGI6Dgq4Uh20RoSmim0IA2LftldQIIzQxaF5W2B45TFBr2QMjGrYAfT99HhClmFeTNEzYeA5o30oKg3A7COKSouy+yF7BFpSWysD2BpoV27ldLIanO9ZDfYbCh9HBkq7c8W56WheiQOIs0JNqLvZcE9lqbZmI6jpMTD7c1gzYV951Eu5R6hODKYaHR2IL6kbAk+IQe1UxVE3b7Yk9hhd+IvvbOdTK7UlTUIPmC3d2+NshbsW1t5CS2M29DZkBhrD0T4pDaBpVtELxsxenYBljNXpb7K73UtjkqOmEU3/EdKA58l51JXJ3uHnz0X1hcRRgkyQFF5g5M3k6g3Kl/a4gbBzh2/3PH4lB3YpDW59W3g9Mk4U5K1Ug83g7J2u/5FF5GmgvYn5Z9e+RESRcos95Hceg4Dp6ni7yYp8ojnxBb/pSJc9GCuqV0aVzXqAGIT4vuIiouO24iBpEZbI0NXVlK/7ZJFIfUvpL9CCw7XnCTb/ZafnHcE3ERRQvOFKaPHK98ZXF9f84X0Hf8FpaMjmRlWUPko/Q8a7XGRrmTF+6/FVvF4L/dUt7XpdDR5Bz+rMH5oqOPToX7grxVWDbzHQS6adE1PThfYswrUPOqGE0wbUAbHjo2tDmHjKv+0aet51nl1fE4kJwJe+Me/dW/rn+o+kfuwuvEw==) to illustrate what I'm saying 
in either case, they should return  type , props, children  

and storyFn()  returns  render function all time, the reason why i put   element.props  because the type definition is FunctionComponent normally i should be getting props from storyFn()  i have no club why they sending it as a Wrapper with render function inside

Reply:
sounds good to me!

Reply:
render() it is not rendering anything it returns a vnode,  virtual dom presentation of the component.